c2corg_ui/static/js/feed.js

```diff
@@ -0,0 +1,121 @@
+goog.provide('app.FeedController');
+goog.provide('app.feedDirective');
+
+goog.require('app');
+goog.require('app.Alerts');
+goog.require('app.Api');
+goog.require('app.Authentication');
+goog.require('app.utils');
+
+
+/**
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.feedDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appFeedController as feedCtrl'
+  };
+};
+app.module.directive('appFeed', app.feedDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {app.Authentication} appAuthentication
+ * @param {app.Alerts} appAlerts
+ * @param {app.Api} appApi Api service.
+ * @param {string} authUrl Base URL of the authentication page.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.FeedController = function($scope, $attrs, appAuthentication, appAlerts,
+    appApi, authUrl) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.auth_ = appAuthentication;
+
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.location_ = $attrs['location'];
+
+  /**
+   * @type {Array<Object>}
+   * @export
+   */
+  this.documents = [];
+
+  switch (this.location_) {
+    case 'home':
+      this.getHomeFeed_();
+      break;
+    case 'user-profile':
+      this.getUserProfileFeed_();
+      break;
+    default:
+      break;
+  }
+};
+
+/**
+ * @private
+ */
+app.FeedController.prototype.getHomeFeed_ = function() {
+  if (this.auth_.isAuthenticated()) {
+    this.api_.readFeed('/personal-feed').then(function(response) {
+      var data = response['data']['feed'];
+      this.documents = data;
+    }.bind(this));
+  } else {
+    this.api_.readFeed('/feed').then(function(response) {
+      var data = response['data']['feed'];
+      this.documents = data;
```

Comment:
This code seems the same than for "personal"? If yes, what about defining a named function?